### PR TITLE
Adds Cluster class to perform operations on cluster within the tests

### PR DESCRIPTION
**Pull request by nshidlin**
The idea is to migrate helper functions from BaseTest to Cluster. 
No breaking changes yet. Subsequent PRs will rebase existing tests to use Cluster object
test_cancel_reset_before_node_boot is rebased to use Cluster object

### CODE REVIEW

File: discovery-infra/test_infra/cluster.py [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+
+class Cluster:
+    
+    def __init__(self, api_client, cluster_name):
+        res = self._create(api_client, cluster_name)
+
+        self.id = res.id
+        self.api_client = api_client
+    
+    def _create(self, api_client, cluster_name):
```
Comment (tsorya) [inline discovery-infra/test_infra/cluster.py:15]: if we have api client in init , why we need to provide it to every function?

Comment (nshidlin) [reply discovery-infra/test_infra/cluster.py:15]: I need to be able override the client for auth tests

File: discovery-infra/test_infra/cluster.py [hunk 2]
```diff
@@ -0,0 +1,237 @@
+import logging
+
+from tests.conftest import env_variables
+from test_infra import consts, utils
+
+
+class Cluster:
```
Comment (tsorya) [inline discovery-infra/test_infra/cluster.py:7]: maybe lets add folder for such sort of objects? not sure what its name but will be nice to have it in folder.
help_classes? maybe

File: discovery-infra/test_infra/cluster.py [hunk 3]
```diff
@@ -0,0 +1,237 @@
...
+    
+    def __init__(self, api_client, cluster_name):
+        res = self._create(api_client, cluster_name)
```
Comment (tsorya) [inline discovery-infra/test_infra/cluster.py:10]: maybe lets allow to provide cluster id and if it is provided don't create one? just run get?

Comment (nshidlin) [reply discovery-infra/test_infra/cluster.py:10]: I don't see the use case for this, if the way to create a cluster is by creating a Cluster obeject. wdty?

File: discovery-infra/test_infra/cluster.py [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+        api_client.generate_and_download_image(
+            cluster_id=self.id,
+            ssh_key=ssh_key,
+            image_path=iso_download_path
+        )
+
+    def wait_until_hosts_are_discovered(
+        self, api_client=None, nodes_count=env_variables['num_nodes']
+    ):
+        if api_client is None:
```
Comment (tsorya) [inline discovery-infra/test_infra/cluster.py:50]: if u need it for some reason, lets create decorator for this one, you can add this line to all functions.

**Pull request merged**